back/engines/commercial/bulk_import_ideas/spec/acceptance/import_ideas_spec.rb

```diff
@@ -36,41 +40,287 @@
+              expect(Idea.all.count).to eq 1
+              expect(BulkImportIdeas::IdeaImport.count).to eq 1
+              expect(BulkImportIdeas::IdeaImportFile.count).to eq 1
+              expect(project.reload.ideas_count).to eq 0 # Draft ideas should not be counted
+
+              # Relationships
+              expect(response_data.first.dig(:relationships, :idea_import, :data)).not_to be_nil
+              expect(json_response_body[:included].pluck(:type)).to include 'idea_import'
+            end
+          end
+
+          context 'timeline projects' do
+            let(:project) { create(:project_with_current_phase) }
+            let(:current_phase) { TimelineService.new.current_phase(project) }
+
+            let(:id) { project.id }
+
+            context 'current phase' do
+              example 'Bulk import ideas from scanned .pdf to current phase' do
+                expect_any_instance_of(BulkImportIdeas::GoogleFormParserService).to receive(:process_upload).and_return(nil)
+                expect_any_instance_of(BulkImportIdeas::GoogleFormParserService).to receive(:raw_text_page_array).and_return(create_project_bulk_import_raw_text_array)
+                do_request
+                assert_status 201
+                expect(response_data.count).to eq 1
+                expect(Idea.all.count).to eq 1
+                expect(BulkImportIdeas::IdeaImport.all.count).to eq 1
+                expect(BulkImportIdeas::IdeaImportFile.all.count).to eq 1
+                expect(Idea.all.first.phases.count).to eq 1
+                expect(Idea.all.first.phases.first).to eq current_phase
+              end
+            end
+
+            context 'specified phase' do
+              let(:phase_id) { project.phases.first.id }
+
+              example 'Bulk import ideas from scanned .pdf to a specified phase' do
+                expect_any_instance_of(BulkImportIdeas::GoogleFormParserService).to receive(:process_upload).and_return(nil)
+                expect_any_instance_of(BulkImportIdeas::GoogleFormParserService).to receive(:raw_text_page_array).and_return(create_project_bulk_import_raw_text_array)
+                do_request
+                assert_status 201
+                expect(response_data.count).to eq 1
+                expect(Idea.all.count).to eq 1
+                expect(BulkImportIdeas::IdeaImport.all.count).to eq 1
+                expect(BulkImportIdeas::IdeaImportFile.all.count).to eq 1
+                expect(Idea.all.first.phases.count).to eq 1
+                expect(Idea.all.first.phases.first).to eq project.phases.first
+                expect(Idea.all.first.phases.first).not_to eq current_phase
+              end
+            end
+          end
+        end
+      end
+    end
+
+    context 'idea import metadata' do
+      get 'web_api/v1/idea_imports/:id' do
+        let(:id) do
+          idea = create(:idea)
+          idea.update! idea_import: create(:idea_import, idea: idea)
```

Comment:
It's sufficient to create `:idea_import` without `idea.update!`. Use `idea.reload` if necessary.